### ISSUE TITLE
`change` callback not always executed when `addTags` is called

### ISSUE THREAD

**Issue by spud** [yairEO/tagify#1236]
## Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

https://jsbin.com/detogoheca/1/edit?html,js,output

I'm trying to update a script of mine from tagify 4.12 to the current (4.17.9), and it doesn't behave the same way.

I create an "onChange" function, and after instantiating Tagify using the onChange function as a "change" callback, I call 

addTags('foo');
removeTag();

_solely_ to trigger the change handler (which creates hidden form fields for saving upon form submission).

In the jsbin posted above, it appears that, when there is **no** default value in the input, the change handler **is** executed. But if I add a comma-delimited default value, the change handler is **not** excuted. (The jsbin starts with a value, but removing the default value will display the other behavior, properly logging in the console.)

Manually adding or deleting a tag — whether or not there is a default value — does properly execute the change handler. 

Version 4.12 did trigger the change event when addTags was called.

In general, is there some way to simply fire the change handler without doing the addTags/remoteTag dance?

**Comment by spud** [comment]
I'm storing the tag values in a database as a PHP serialized array, so I want my form to submit the values using the typical

`name="role[]"` 

format commonly used for submitting an array of values.

My onChange handler for Tagify basically tells Tagify to create a hidden field for each tag in the input, e.g.

`<input type="hidden" name="role[]" value="Subscriber" />`
`<input type="hidden" name="role[]" value="Patron" />`

This works perfectly when starting from a blank input. Each tag I add creates a hidden field, and those get submitted in the end.

When I load the form and the input has _existing_ values, however, the field is initially set up like this:

`<input class="f_tags" type="text" name="role" value="Anonymous,User,Editor">`

Tagify then tagifies the input, but to properly submit the values, I need Tagify to first **re-create the same hidden fields** for the existing values that it creates for new tags.

With Tagify 4.12, I could call

`tagify.addTags('&nbsp;'); 
tagify.removeTag();`

to trigger the `change` handler. Tagify would execute my onChange function, which creates the hidden fields. 

With 4.17.9, that no longer happens, so unless I _manually_ edit the tag input, my form won't submit the values as an array (which the processor expects).

Basically, I need a way to programmatically execute the Tagify change handler on demand, immediately after initializing the field, in order to handle existing values.

If there is some fundamentally different approach to achieve my goals, I'd love to hear it. Otherwise, Tagify is unusable in its current version (for my needs).

In the most relevant Git post about this process, you state 

> I prefer the developer who implements the solution to take care of value format transformation in their end rather than implementing some solution within Tagify to customize the value's format.

…and so I wrote a change handler to do just that, but I need to be able to trigger it when I render the input!

**Comment by yairEO** [comment]
> My onChange handler for Tagify basically tells Tagify to create a hidden field for each tag in the input, e.g.

Not a good idea at all. Please do not do that. work with a the single `input` element already existing and post its value to the server and parse it. Alternatively send the data in an AJAX post request at any format you want it to be at. use `tagify.value` array, and create from it the desired format
